ARTstor.js

```diff
@@ -1,131 +1,713 @@
+
 {
-	"translatorID": "5278b20c-7c2c-4599-a785-12198ea648bf",
-	"label": "ARTstor",
-	"creator": "Sebastian Karcher",
-	"target": "^https?://library\\.artstor\\.org",
-	"minVersion": "3.1",
-	"maxVersion": "",
-	"priority": 100,
-	"inRepository": true,
-	"translatorType": 4,
-	"browserSupport": "gcs",
-	"lastUpdated": "2014-11-13 15:09:17"
-}
-
-/*
-	***** BEGIN LICENSE BLOCK *****
-	
-	ARTstor Translator, Copyright © 2012 Sebastian Karcher
-	This file is part of Zotero.
-	
-	Zotero is free software: you can redistribute it and/or modify
-	it under the terms of the GNU Affero General Public License as published by
-	the Free Software Foundation, either version 3 of the License, or
-	(at your option) any later version.
-	
-	Zotero is distributed in the hope that it will be useful,
-	but WITHOUT ANY WARRANTY; without even the implied warranty of
-	MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
-	GNU Affero General Public License for more details.
-	
-	You should have received a copy of the GNU Affero General Public License
-	along with Zotero.  If not, see <http://www.gnu.org/licenses/>.
-	
-	***** END LICENSE BLOCK *****
-*/
-
-function getData(doc, checkOnly) {
-	var data = {},
-		widgets = doc.getElementsByClassName('MetaDataWidget'),
-		topWidget = null, topZIndex;
-	
-	// Dtermine the top metadata window
-	for (var i=0; i<widgets.length; i++) {
-		
-		var zindex = doc.defaultView.getComputedStyle(widgets[i], null)
-			.getPropertyValue('z-index');
-		if (!zindex && zindex !== 0) continue;
-		
-		if (!topWidget || topZIndex < zindex) {
-			topWidget = widgets[i];
-			topZIndex = zindex;
-		}
-	}
-	
-	if (!topWidget) return false;
-	
-	var fields = topWidget.getElementsByClassName('field'),
-		found = false;
-	for (var i=0; i<fields.length; i++) {
-		var field = ZU.trimInternal(fields[i].textContent);
-		
-		var value = fields[i].nextElementSibling;
-		if (!value || !value.classList.contains('data')) continue;
-		value = ZU.unescapeHTML(ZU.cleanTags(value.innerHTML));
-		
-		if (!field || !value.trim()) continue;
-		
-		if (checkOnly) return true;
-		
-		found = true;
-		data[field] = value;
-	}
-	
-	return found ? data : false;
+    "translatorID": "5278b20c-7c2c-4599-a785-12198ea648bf",
+    "label": "ARTstor",
+    "creator": "John Justin, Charles Zeng",
+    "target": "\.artstor|\.sscommons\.org:?\w*\/(open)*library",
```

Comment:
You never replied to aurimas's comment&suggested target above: https://github.com/zotero/translators/pull/899#discussion_r33820499 which I do think would be the best way forward.

If ```.artstor|.sscommons.org:?\w\/(open)library``` passes and ```\.artstor|\.sscommons.org:?\w\/(open)library``` fails in Scaffold, that means your regex is broken: the first one uses the dot as a metacharacter for any character, the second one escapes it to a literal period. ```\.``` definitely works in Scaffold, you can look at any of the other 300+ translators.

What I was referring to with the "or" marker (|) is that you place it, without any parentheses, between \.artstor and the rest of the regex, which means the regex will evaluate to true with any string containing ".artstor"

But again, instead of fixing&discussing these piecemeal, why don't you just tell us for which URLs ```"https?://([^/]+\\.)?(artstor|sscommons)\\.org\/(open)?library"``` doesn't work and we'll take it from there.